File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/server/ServerGameTime.java
package pize.tests.voxelgame.server;

import pize.tests.voxelgame.main.time.GameTime;
import pize.tests.voxelgame.main.net.packet.CBPacketTime;

public class ServerGameTime extends GameTime{
    
    private final Server server;
    
    public ServerGameTime(Server server){
        this.server = server;
    }
    
    public Server getServer(){
        return server;
    }
    
    
    @Override
    public void setTicks(long ticks){
        super.setTicks(ticks);
        server.getPlayerList().broadcastPacket(new CBPacketTime(ticks));
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/server/ServerGameTime.kt
package pize.tests.voxelgame.server

import pize.tests.voxelgame.main.net.packet.CBPacketTime
import pize.tests.voxelgame.main.time.GameTime

class ServerGameTime(val server: Server) : GameTime() {

    override var ticks: Long
        get() = super.ticks
        set(ticks) {
            super.setTicks(ticks)
            server.playerList.broadcastPacket(CBPacketTime(ticks))
        }
}
